lib/logical.js

```diff
@@ -63,23 +64,27 @@ function logicalTree(fileTree, options) {
         return false; // this was from a dev dep, so let's lose it
       }
 
-      dep.extraneous = true;
-      dep.depType = depTypes.EXTRANEOUS;
-      var issue = ext + ': ' + dep.name + '@' + dep.version + ' (from ' +
-        dep.dep + ') > ' + dep.__filename;
-      dep.problems = [issue];
-      problem(logicalRoot, issue);
-      insertLeaf(logicalRoot, dep, fileTree);
+      var leaf = copy(dep);
+
+      var issue = format('%s: %s@%s (from %s) > %s', ext, leaf.name,
+        leaf.version, leaf.dep, path.relative('.', leaf.__filename));
+      leaf.problems = [issue];
+      problems.push(issue);
+      leaf.extraneous = true;
+      leaf.depType = depTypes.EXTRANEOUS;
+      leaf.dependencies = walkDeps(fileTree, dep);
```

Comment:
This is a change in the extraneous packages. Before it would show the extraneous packages as a stand alone leaf, but they do actually have dependencies, so this change copies the dep (so we don't mess with the original physical tree) and then walks the deps).